radio/src/gui/colorlcd/radio_setup.cpp

```diff
@@ -48,151 +48,163 @@ class DateTimeWindow : public FormGroup {
     {
       FormGroup::checkEvents();
 
-      if (get_tmr10ms() - lastRefresh >= 10) {
-        struct gtm newTS;
+      if (seconds && (get_tmr10ms() - lastRefresh >= 10)) {
         lastRefresh = get_tmr10ms();
 
-        gettime(&newTS);
-        if(newTS.tm_sec == lastSecond)
-          return;
-
-        lastSecond = newTS.tm_sec;
-        year->update();
-        month->update();
-        day->update();
-        hour->update();
-        minutes->update();
-        seconds->update();
+        gettime(&m_tm);
+        if (m_tm.tm_year != m_last_tm.tm_year) {
+          m_last_tm.tm_year = m_tm.tm_year;
+          year->update();
+        }
+        if (m_tm.tm_mon != m_last_tm.tm_mon) {
+          m_last_tm.tm_mon = m_tm.tm_mon;
+          month->update();
+        }
+        if (m_tm.tm_mday != m_last_tm.tm_mday) {
+          m_last_tm.tm_mday = m_tm.tm_mday;
+          day->update();
+        }
+        if (m_tm.tm_hour != m_last_tm.tm_hour) {
+          m_last_tm.tm_hour = m_tm.tm_hour;
+          hour->update();
+        }
+        if (m_tm.tm_min != m_last_tm.tm_min) {
+          m_last_tm.tm_min = m_tm.tm_min;
+          minutes->update();
+        }
+        if (m_tm.tm_sec != m_last_tm.tm_sec) {
+          m_last_tm.tm_sec = m_tm.tm_sec;
+          seconds->update();
```

Comment:
I checkt the numberedit before making my last changes here and it does compare the old and the new value in the update function.
What I somehow missed, was that it still does the invalidation/redraw :(